### PR TITLE
Validate patch requests with JSON schema

### PR DESCRIPTION
**Pull request by agger-magenta**

### CODE REVIEW

**Comment by andreaskring** [comment]
@agger-magenta Looks good to me - I only have a single concern to raise: no new tests have been added to the `test_validate.py` that tests the `do_create=False` case. I think you should add a few tests that takes care of this use case, but it is of course up to you :-)

**Comment by andreaskring** [comment]
(I have not run the test suite)

**Comment by agger-magenta** [comment]
> I only have a single concern to raise: no new tests have been added to the test_validate.py that tests the do_create=False case. I think you should add a few tests that takes care of this use case, but it is of course up to you 

Yes, that makes sense. I'll add a couple of tests.

File: oio_rest/oio_rest/validate.py [hunk 1]
```diff
@@ -419,26 +436,40 @@ def generate_json_schema(obj):
     return schema
 
 
+# Store already calculated schemas to save performance.
```
Comment (agger-magenta) [inline oio_rest/oio_rest/validate.py]: This `StoredSchema` class is dead wood and should be removed.

Comment (cmoesgaard) [reply oio_rest/oio_rest/validate.py]: I'll remove it.

**Pull request merged**